rotkehlchen/db/upgrades/v36_v37.py

```diff
@@ -185,33 +187,131 @@ def _update_ens_mappings_schema(write_cursor: 'DBCursor') -> None:
+                str(-amount) if amount != ZERO else str(amount),
+                str(-usd_value) if usd_value != ZERO else str(usd_value),
                 'Automatic virtual conversion of staked ETH rewards to ETH',
                 event_row[0],
             ),
         )
-
     if len(update_tuples) != 0:
         write_cursor.executemany(
             'UPDATE history_events SET type=?, subtype=?, amount=?, usd_value=?, notes=? WHERE identifier=?',  # noqa: E501
             update_tuples,
         )
-    log.debug('Exit _fix_kraken_eth2_events')
+
+    log.debug('Fixing kraken trades')
+    update_tuples = []
+    write_cursor.execute(
+        'SELECT identifier, amount, usd_value FROM history_events WHERE location="B" AND '
+        'type="trade" AND subtype="none"',
+    )
+    for event_row in write_cursor:
+        asset_amount = FVal(event_row[1])
+        usd_value = FVal(event_row[2])
+        update_tuples.append(
+            (
+                HistoryEventType.TRADE.serialize(),
+                HistoryEventSubType.SPEND.serialize() if asset_amount < ZERO else HistoryEventSubType.RECEIVE.serialize(),  # noqa: E501
+                str(-asset_amount) if asset_amount < ZERO else str(asset_amount),
+                str(-usd_value) if usd_value < ZERO else str(usd_value),
+                event_row[0],
+            ),
+        )
+
+    if len(update_tuples) != 0:
+        write_cursor.executemany(
+            'UPDATE history_events SET type=?, subtype=?, amount=?, usd_value=? WHERE identifier=?',  # noqa: E501
+            update_tuples,
+        )
+
+    log.debug('Fixing kraken withdrawals')
+    update_tuples = []
+    write_cursor.execute(
+        'SELECT identifier, amount, usd_value FROM history_events WHERE location="B" AND '
+        'type="withdrawal"',
+    )
+    for event_row in write_cursor:
+        amount = FVal(event_row[1])
+        usd_value = FVal(event_row[2])
+        update_tuples.append(
+            (
+                str(-amount) if amount != ZERO else str(amount),
+                str(-usd_value) if usd_value != ZERO else str(usd_value),
+                event_row[0],
+            ),
+        )
+    if len(update_tuples) != 0:
+        write_cursor.executemany(
+            'UPDATE history_events SET amount=?, usd_value=? WHERE identifier=?',
+            update_tuples,
+        )
+
+    log.debug('Fixing kraken instant swaps')
+    update_tuples = []
+    grouped_events: dict[str, list[Any]] = defaultdict(list)
+    write_cursor.execute(
+        'SELECT identifier, event_identifier, type, amount, usd_value FROM history_events '
+        'WHERE location="B" AND type IN ("spend", "receive") AND subtype="none"',
+    )
+    for row in write_cursor:
+        grouped_events[row[1]].append(row)
+
+    for groups in grouped_events.values():
+        if len(groups) != 2:
+            continue
+
+        for event in groups:
+            event_type = HistoryEventType.TRADE.serialize()
```

Comment:
What is the reason for doing this here? Why not to do it when appending to `update_tuples`?